nslsii/__init__.py

```diff
@@ -160,14 +160,29 @@ def configure_base(user_ns, broker_name, *,
         from ophyd import setup_ophyd
         setup_ophyd()
 
-    if not ophyd_logging:
-        # Turn on error-level logging, particularly useful for knowing when
-        # pyepics callbacks fail.
+    if configure_logging:
         import logging
-        import ophyd.ophydobj
-        ch = logging.StreamHandler()
-        ch.setLevel(logging.ERROR)
-        ophyd.ophydobj.logger.addHandler(ch)
+        from logging.handlers import TimedRotatingFileHandler
+
+        log_file_handler = TimedRotatingFileHandler(
+            filename=os.path.expanduser("/var/log/bluesky/bluesky.log"),
+            when="W0",
+            backupCount=10
+        )
+        log_file_handler.setLevel("INFO")
+
+        logging.getLogger("bluesky").addHandler(log_file_handler)
+        logging.getLogger("bluesky.*").addHandler(log_file_handler)
+        logging.getLogger("ophyd").addHandler(log_file_handler)
+        logging.getLogger("ophyd.*").addHandler(log_file_handler)
+
+        console_handler = logging.StreamHandler()
```

Comment:
This would produce WARNING logs _in addition to_ any log messages issued by the user-facing handler managed by the `config_X_logging` convenience functions. In the past we have received some push-back from beamline staff that do not find WARNING or even ERROR messages helpful, so it's important to have an easy way to turn the volume  #down.

I support pushing out WARNING-level console logging by default --- it seems like it could help us spot some latent problems --- but I think using the `config_X_logging` functions might be the way to go so that staff can use those "easy mode" functions to adjust the volume as needed.

Reply:
Ok.